### PR TITLE
Docs/general enhancements

### PR DESCRIPTION
**Pull request by PipKat**
This PR adds the deprecation best practices information that is currently in the ansys_sphinx_theme repo per Issue 145 and also adds Vale information to the _PyAnsys Developer's Guide_. Once this PR is merged, I will remove the deprecation best practices information from the doc for ansys_sphinx_theme. Also, once this PR is merged, would someone please regenerate this guide. It was last regenerated on March 25!

### CODE REVIEW

File: doc/source/coding-style/deprecation.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+.. code:: python
+
+    class DeprecationError(RuntimeError):
+        """Used for depreciated methods and functions."""
+
+        def __init__(self, message='This feature has been depreciated.'):
+            """Empty init."""
+            RuntimeError.__init__(self, message)
+
+You then use his custom ``DeprecationError`` in place of an ``Exception``.
```
Comment (RobPasMue) [inline doc/source/coding-style/deprecation.rst:77]:
```suggestion
You then use this custom ``DeprecationError`` in place of an ``Exception``.
```

File: doc/source/coding-style/deprecation.rst [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            # or raise an AttributeError (could also make a custom DeprecationError)
+            raise AttributeError('`assignmaterial` is deprecated. Use `assign_material` instead.')
+
+        def assign_material(self, obj, mat):
+            """Assign a material to one or more objects.
+            ...
+            """
+
+
+If you remove a method entirely, there's no reason to provide a link to the old
```
Comment (RobPasMue) [inline doc/source/coding-style/deprecation.rst:51]:
```suggestion
If you remove a method entirely, there is no reason to provide a link to the old
```

File: doc/source/coding-style/deprecation.rst [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+- Have the old method call the new method and raise a warning
+- Raise an ``AttributeError`` if you remove the method entirely
+
+In the docstring of the old method, provide a `Sphinx Deprecated Directive
+<https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-deprecated>`_
+that links to the new method. This way, you notify your users when you make
+an API change and give them a chance to change their code. Otherwise,
+users stop upgrading, or worse, stop using your API, due to stability concerns.
+For this reason, it's best to use a warning first and then use an error after
```
Comment (RobPasMue) [inline doc/source/coding-style/deprecation.rst:20]:
```suggestion
For this reason, it is best to use a warning first and then use an error after
```

File: doc/source/doc-style/formatting-tools.rst [hunk 4]
```diff
@@ -133,12 +127,51 @@ it under the ``[tool.pydocstyle]`` entry:
...
+
+.. code-block:: python
+
+   vale .
+
+Check all files in the repository with:
+
+.. code-block:: python
+
+   vale doc --config=doc/.vale.ini
```
Comment (RobPasMue) [inline doc/source/doc-style/formatting-tools.rst:158]:
```suggestion
   vale --config=doc/.vale.ini .
```

File: doc/source/doc-style/formatting-tools.rst [hunk 5]
```diff
@@ -133,12 +127,51 @@ it under the ``[tool.pydocstyle]`` entry:
...
+
+   vale sync
+
+Check all files in the ``doc`` folder with:
+
+.. code-block:: python
+
+   vale .
+
+Check all files in the repository with:
```
Comment (RobPasMue) [inline doc/source/doc-style/formatting-tools.rst:154]:
```suggestion
Check all files in the repository, by going to the ``root`` directory and running:
```

File: doc/source/doc-style/formatting-tools.rst [hunk 6]
```diff
@@ -133,12 +127,51 @@ it under the ``[tool.pydocstyle]`` entry:
...
+
+.. code-block:: python
```
Comment (RobPasMue) [inline doc/source/doc-style/formatting-tools.rst:156]:
```suggestion
.. code-block:: bash
```

File: doc/source/doc-style/formatting-tools.rst [hunk 7]
```diff
@@ -133,12 +127,51 @@ it under the ``[tool.pydocstyle]`` entry:
...
+
+In the library's ``doc`` folder, download the package with:
+
+.. code-block:: python
+
+   vale sync
+
+Check all files in the ``doc`` folder with:
+
+.. code-block:: python
```
Comment (RobPasMue) [inline doc/source/doc-style/formatting-tools.rst:150]:
```suggestion
.. code-block:: bash
```

File: doc/source/doc-style/formatting-tools.rst [hunk 8]
```diff
@@ -133,12 +127,51 @@ it under the ``[tool.pydocstyle]`` entry:
...
+`Google developer documentation style guide <https://developers.google.com/style/>`_,
+along with any custom Ansys rules and terminology lists, to reStructuredText (RST)
+and Markdown (MD) files.
+
+After a PyAnsys team member implements ``Vale`` in your PyAnsys library, you can check
+any content changes that you make in supported files locally.
+
+In the library's ``doc`` folder, download the package with:
+
+.. code-block:: python
```
Comment (RobPasMue) [inline doc/source/doc-style/formatting-tools.rst:144]:
```suggestion
.. code-block:: bash
```

File: doc/source/doc-style/formatting-tools.rst [hunk 9]
```diff
@@ -133,12 +127,51 @@ it under the ``[tool.pydocstyle]`` entry:
    convention = "numpy"
 
 
+Vale
+----
+`Vale` is a tool for maintaining a consistent style and voice in your documentation.
```
Comment (RobPasMue) [inline doc/source/doc-style/formatting-tools.rst:132]:
```suggestion
`Vale`_ is a tool for maintaining a consistent style and voice in your documentation.
```

**Review by RobPasMue — APPROVED** [review]
Apart from my recommendations, LGTM! I'll generate a new docs version as soon as it's merged :)

**Comment by RobPasMue** [comment]
Since they are mostly editorial comments, I'll merge them to the branch and try to close this PR asap - that way we can make the release right away. Any comments @jorgepiloto?

File: doc/source/doc-style/formatting-tools.rst [hunk 10]
```diff
@@ -25,12 +24,11 @@ this tool by running:
 
 Codespell
 ---------
-
 `Codespell`_ checks for common misspellings in text files. This implies that it
 is not limited to Python files but can run checks on any human-readable file.
 
 It is possible to ignore words that are flagged as misspelled. You can specify these words in a
-file that can hen be passed to `Codespell` by running:
+file that can then be passed to `Codespell` by running:
```
Comment (jorgepiloto) [inline doc/source/doc-style/formatting-tools.rst:31]: Good catch, @PipKat 👍🏽

**Review by jorgepiloto — APPROVED** [review]
Thanks for addressing this, @PipKat 🚀

**Pull request merged**